### ISSUE TITLE
Open a new pane with the environment vars of the parent

### ISSUE THREAD

**Issue by zadjii-msft** [microsoft/terminal#4259]
@zadjii-msft I'm trying to clone the current terminal into a new pane. I need the same environment var's etc carried over similar to how 'start' does it in cmd. I know it isn't currently possible but is there a temporary workaround?

_Originally posted by @Surya-06 in https://github.com/microsoft/terminal/issues/1756#issuecomment-575134471_

**Comment by DHowett-MSFT** [comment]
I'm pulling Triage, but this might not be possible.

**Comment by Surya-06** [comment]
@zadjii-msft @DHowett-MSFT  Now that the duplicate pane option is in, do you guys think it's possible to get the env vars from that pane ? If so I'd really like to help with it cause it's particularly a pain point for me. Let me know if I can help. Thanks !

**Comment by zadjii-msft** [comment]
I honestly don't think it's possible to get the environment variables of a child process trivially. Plus, if there's a whole tree of processes running in a tab, then which process's environment variables should be the ones duplicated? I'd reckon that this counts as a _hard_ problem

**Comment by Surya-06** [comment]
Yes, it doesn't seem pretty straight forward to bring the env vars from another process and set them here but I'm seriously interested in making this happen cause this would be a unique thing in Terminal different from every other application. If you can give me some pointers I would at least like to explore the possibility. I also think this isn't so walled off since conemu does it ( duplicate root process option). This integrated with duplicate tab would seriously make duplication more useful. Thoughts ? @zadjii-msft

**Comment by Surya-06** [comment]
I also just realized another alternative, can we allow users to configure env vars when opening profiles. This will allow them to set the vars and just duplicate the profile and everything should still be fine right ?

**Comment by piotrpalek** [comment]
Hey I've found this issue because I was searching for the following: I'm not sure how it's for other people, but for me it 99% of what I want is to have the split open the same shell in the current path I'm in, would that be something that's doable?

I know there's a PR already in the works for opening a split with the current shell, what about the path?

**Comment by zadjii-msft** [comment]
@piotrpalek You're going to want to take a look at #3158

**Comment by AlexxxRu** [comment]
@zadjii-msft @Surya-06 
Maybe you can get some clues from ConEmu sources. This application has the feature to duplicate the tab with the same env and CWD. Actually, only the lack of this feature stops me from using Windows Terminal.

**Comment by skyline75489** [comment]
@AlexxxRu Funny that you mentioned ConEmu, the CWD part is implemented precisely in the same way as ConEmu. See https://github.com/microsoft/terminal/issues/3158#issuecomment-764003033 and #8330

**Comment by AlexxxRu** [comment]
@skyline75489 Yes, looks so, but env variables are still lost.

**Comment by zadjii-msft** [comment]
Hmm. Still don't know a good way to do this. But looking through https://conemu.github.io/en/ConEmuEnvironment.html#Export_variables

Maybe the clever workaround would be to run the (hypothetical) `wt -w 0 new-tab --duplicate` or something. By running the new process, it would inherit the current env block, and then the duplicated tab _should_ start with the same variables. (admittedly, we'd still need to transit the env vars through the `IPeasant.ExecuteCommandline` method, but that's an implementation detail).

I'd still love a more seamless way to do this though that doesn't necessarily involve cooperation from the shell launching a new process, but that might be **H**ard
